### PR TITLE
[Fix #2009] Use SafeYAML.whitelist! to allow deserialize Regexp when SafeYAML is required

### LINKED ISSUE #2009: safe_yaml vs rubocop
In my rails project when i do rake rubocop it fails with ```An error occurred while Style/WordArray cop was inspecting ...```. When i do plain rubocop -d it work fine.

Some debugging showed this: My rails project uses rails_admin which loads safe_yaml. It looks safe_yaml disables regex loading of ``` !ruby/regexp``` from yaml by default.
```ruby
require 'yaml'
require 'safe_yaml'
YAML.safe_load("foo: !ruby/regexp '/bar/'\n",[Regexp])
```

Seems to be caused by this commit
https://github.com/bbatsov/rubocop/commit/36b857f465906e76f9a8f76e98820ee3c6be2600

### PR DESCRIPTION
**Pull request by eitoball**
`SafeYAML` overrides `YAML.safe_yaml` with `SafeYAML.load`. This method has different method signature.

### CODE REVIEW

File: spec/rubocop/config_loader_spec.rb [hunk 1]
```diff
@@ -291,6 +291,27 @@
...
+        create_file(configuration_path, [
+          'Style/WordArray:',
+          "  WordRegex: !ruby/regexp '/\\A[\\p{Word}]+\\z/'"
+        ])
+      end
+
+      it 'de-serializes Regexp class' do
+        # Load SafeYAML in different memory space
+        pid = ::Process.fork {
+          require 'safe_yaml'
```
Comment (bbatsov) [inline spec/rubocop/config_loader_spec.rb]: The only problem I see is that after this point `safe_yaml` will be loaded for the rest of the specs.

**Comment by bbatsov** [comment]
Apart from the failing build - please, capitalise the word "use" in the commit message's title.

File: spec/rubocop/config_loader_spec.rb (hunk 1 above)
Comment (eitoball) [inline spec/rubocop/config_loader_spec.rb]: I think that `safe_yam` is loaded only in `Process.fork`.

```ruby
Process.fork do
  require 'yam'
  defined?(YAML) #=> "constant"
end
Process.waitall
defined?(YAML) #=> false
```

**Comment by eitoball** [comment]
> Apart from the failing build - please, capitalise the word "use" in the commit message's title.

Fixed commit message. Build and coverage are fine now. Thanks.

**Comment by bbatsov** [comment]
I see you've had to make some changes to the coveralls setup. I think it'd be best if those were in a separate commit.

**Comment by eitoball** [comment]
Separated into 2 commits; one is for safe_yaml fix, and another is for coveralls setup. How are they? Thank you for reviewing.

**Comment by bbatsov** [comment]
Looks good. Thanks!

**Pull request merged**